addons/web/static/tests/views/form/form_view_tests.js

```diff
@@ -9389,6 +9389,61 @@ QUnit.module("Views", (hooks) => {
         );
     });
 
+    QUnit.test(
+        'buttons with "confirm-title" and "confirm-label" attributes save before calling the method',
+        async function (assert) {
+            const actionService = {
+                start() {
+                    return {
+                        async doActionButton(args) {
+                            assert.step("execute_action");
+                        },
+                    };
+                },
+            };
+            registry.category("services").add("action", actionService, { force: true });
+
+            await makeView({
+                type: "form",
+                resModel: "partner",
+                serverData,
+                arch: `
+                    <form>
+                        <header>
+                            <button name="post" class="p" string="Confirm" type="object" confirm="Very dangerous. U sure?"
+                                confirm-title="Confirmation" confirm-label="Confirm"/>
```

Comment:
I would use other word than "Confirmation" because it's the default one, so we don't know if it's correctly set because of your option of because of the default value of the confirmation dialog
```suggestion
                                confirm-title="Confirm Title" confirm-label="Confirm Label"/>
```